app/controllers/profiles_controller.rb

```diff
@@ -7,13 +7,29 @@ def index
   end
 
   def show
-    profile = Profile.find(params[:id])
     authorize profile
     render json: ProfileSerializer.new(profile)
   end
 
+  def tailoring_file
+    return unless profile.tailored?
+
+    send_data XccdfTailoringFile.new(
+      profile: profile,
+      rule_ref_ids: profile.tailored_rule_ref_ids
+    ).to_xml, filename: tailoring_filename, type: Mime[:xml]
+  end
+
   private
 
+  def tailoring_filename
+    "#{profile.benchmark.ref_id}__#{profile.ref_id}__tailoring.xml"
+  end
+
+  def profile
+    @profile ||= Profile.find(params[:id])
```

Comment:
Can we change this one to `Pundit.policy_scope(current_user, ::Profile).find(params[:id])` and remove the `authorize` line? No need to do it in two steps and search in a larger table when we can restrict it to user owned profiles

Reply:
This PR addresses your comment: https://github.com/RedHatInsights/compliance-backend/pull/347